proof/invariant-abstract/IpcDet_AI.thy

```diff
@@ -2544,13 +2539,13 @@ lemma rai_invs':
      apply (clarsimp simp: idle_no_ex_cap)
     apply (wpsimp simp: do_nbrecv_failed_transfer_def wp: valid_irq_node_typ)
    apply (case_tac is_blocking; simp)
-    apply (wpsimp simp: invs_def valid_state_def valid_pspace_def valid_ntfn_def
-                        live_def live_ntfn_def do_nbrecv_failed_transfer_def
-                        valid_tcb_state_def
-                    wp: maybe_return_sc_schedule_tcb sort_queue_ntfn_bound_tcb sts_only_idle
-                        sts_valid_replies sts_fault_tcbs_valid_states
+    apply (clarsimp cong: conj_cong
+           | wpsimp wp: maybe_return_sc_schedule_tcb  sts_only_idle sts_valid_replies
+                        sts_fault_tcbs_valid_states valid_ioports_lift
                         valid_irq_node_typ[where f="as_user t f" for t f]
-                        valid_ioports_lift)
+                  simp: invs_def valid_state_def valid_pspace_def valid_tcb_state_def
+                        valid_ntfn_def live_ntfn_def live_def do_nbrecv_failed_transfer_def
+           | rule_tac R="\<lambda>_ _. ntfn_bound_tcb notification = None" in hoare_post_add)+
```

Comment:
I'm a bit worried about this `hoare_post_add`. If something changes and this step doesn't solve the hoare triples this will loop won't it?